lib/compat/wordpress-6.0/class-wp-webfonts.php

```diff
@@ -78,24 +96,47 @@ public function get_providers() {
 	/**
 	 * Register a webfont.
 	 *
-	 * @param array $font The font arguments.
+	 * @param string $id The unique identifier for the webfont.
+	 * @param array  $font The font arguments.
 	 */
-	public function register_font( $font ) {
+	public function register_font( $id, $font ) {
+		if ( ! $id ) {
+			trigger_error( __( 'An ID is necessary when registering a webfont.', 'gutenberg' ) );
+			return false;
+		}
+
 		$font = $this->validate_font( $font );
 		if ( $font ) {
-			$id                    = $this->get_font_id( $font );
-			self::$webfonts[ $id ] = $font;
+			self::$registered_webfonts[ $id ] = $font;
 		}
 	}
 
 	/**
-	 * Get the font ID.
+	 * Enqueue a webfont.
 	 *
-	 * @param array $font The font arguments.
-	 * @return string
+	 * @param string     $id The unique identifier for the webfont.
+	 * @param array|null $font The font arguments.
 	 */
-	public function get_font_id( $font ) {
-		return sanitize_title( "{$font['font-family']}-{$font['font-weight']}-{$font['font-style']}-{$font['provider']}" );
+	public function enqueue_font( $id, $font = null ) {
+		if ( ! $id ) {
+			trigger_error( __( 'An ID is necessary when enqueueing a webfont.', 'gutenberg' ) );
+			return false;
+		}
+
+		if ( $font ) {
+			$font                           = $this->validate_font( $font );
+			self::$enqueued_webfonts[ $id ] = $font;
+
+			if ( isset( self::$registered_webfonts[ $id ] ) ) {
+				unset( self::$registered_webfonts[ $id ] );
+			}
+		} elseif ( isset( self::$registered_webfonts[ $id ] ) ) {
+			self::$enqueued_webfonts[ $id ] = self::$registered_webfonts[ $id ];
```

Comment:
Reference the variable as an instance member because that's what it is.


```suggestion
			$this->enqueued_webfonts[ $id ] = $this->registered_webfonts[ $id ];
```